services/worker/src/worker/main.py

```diff
@@ -1,53 +1,90 @@
-# SPDX-License-Identifier: Apache-2.0
-# Copyright 2022 The HuggingFace Authors.
+import json
+import logging
+import os
+import sys
+import tempfile
+import time
+from typing import Optional
 
+from filelock import FileLock
 from libcommon.log import init_logging
-from libcommon.processing_graph import ProcessingGraph
-from libcommon.resources import CacheMongoResource, QueueMongoResource
-from libcommon.storage import init_assets_dir
+from libcommon.queue import Job, Status, get_datetime
+from libcommon.resources import QueueMongoResource
+from mirakuru import OutputExecutor
 
+from worker import start_worker_loop
 from worker.config import AppConfig
-from worker.job_runner_factory import JobRunnerFactory
-from worker.loop import Loop
-from worker.resources import LibrariesResource
+from worker.loop import WorkerState
+
+WORKER_STATE_FILE_NAME = "worker_state.json"
+START_WORKER_LOOP_PATH = start_worker_loop.__file__
+
+
+class WorkerExecutor:
```

Comment:
Could we move that to its own file? so that main.py remains as easy to understand as possible?